stacks/api/bucket/bucket.controller.ts

```diff
@@ -262,4 +266,73 @@ export class BucketController {
       });
     });
   }
+
+  async clearRemovedFields(
+    bucketDataService: BucketDataService,
+    previousSchema: Bucket,
+    currentSchema: Bucket
+  ) {
+    let removedKeys = this.findRemovedKeys(
+      previousSchema.properties,
+      currentSchema.properties,
+      [],
+      ""
+    );
+    if (removedKeys.length < 1) return;
+
+    let unsetFields = removedKeys.reduce((pre, acc: any, index, array) => {
+      acc = {...pre, [array[index]]: ""};
+      return acc;
+    }, {});
+
+    await bucketDataService.updateMany(previousSchema._id, {}, {$unset: unsetFields});
+  }
```

Comment:
Can you move these helpers into a new module?

Reply:
Sure.